### DISCUSSION TITLE
Re-export ErrorResponse

### DISCUSSION THREAD

**Discussion by kentcdodds** [remix-run/remix#7238] [Proposals]
The `ErrorResponse` type is in `@remix-run/router`, but I don't think the intent is for people to manually install that package. So could we get it exported from `@remix-run/react` or something so we can use it without having to install `@remix-run/router` and making sure that version stays in sync?

**Comment by brophdawg11** [comment]
Yeah this has come up before.  It's specifically not "publicly" re-exported through Remix at the moment because it's a `class` and we do not want people trying to do something like `throw new ErrorResponse(...)` since the class itself exposes implementation details around user-thrown Responses from loaders/actions and internal-router-thrown "Error" Responses (400, 403, 405, etc.).  It's exported from `@remix-run/router` but marked `@private` since we need access to it in the Remix codebase.

I agree we should export an `interface` of some sort that can be public so folks have a way to type these as props/function params etc.

**Reply by kentcdodds** [reply]
Makes sense.

> so folks have a way to type these as props/function params etc.

Yep, that's what I need it for 👍

**Reply by brophdawg11** [reply]
😍  TIL about `InstanceType<>`: https://github.com/remix-run/react-router/pull/10811

**Reply by kentcdodds** [reply]
Oh brilliant!

**Comment by brophdawg11** [comment]
Fixed by https://github.com/remix-run/remix/pull/7244 and will be available in Remix v2

**Reply by kentcdodds** [reply]
Awesome, thank you!
